zaza/utilities/juju.py

```diff
@@ -533,7 +533,10 @@ def get_application_ip(application, model_name=None):
     :rtype: str
     """
     if is_k8s_deployment():
-        ip = get_k8s_ingress_ip(application, model_name=model_name)
+        try:
+            ip = get_k8s_ingress_ip(application, model_name=model_name)
+        except KeyError:
```

Comment:
can we get a log line?, so this behavior leaves a trace, in the past chasing down "why foo didn't set the IP correctly?" has been time consuming.

Reply:
depending on how often you expect this happen, you could use `logging.exception` - https://docs.python.org/3/library/logging.html#logging.exception